product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/ui/WalletConnectModalViewModel.kt

```diff
@@ -60,17 +64,26 @@ internal class WalletConnectModalViewModel(
 
     private suspend fun createModalState(uri: String) {
         try {
-            val wallets = if (WalletConnectModal.recommendedWalletsIds.isEmpty()) {
+            wallets = if (WalletConnectModal.recommendedWalletsIds.isEmpty()) {
                 getWalletsUseCase(sdkType = WCM_SDK, chains = chains, excludedIds = WalletConnectModal.excludedWalletsIds)
             } else {
-                getWalletsUseCase(sdkType = WCM_SDK, chains = chains,  excludedIds = WalletConnectModal.excludedWalletsIds, recommendedIds = WalletConnectModal.recommendedWalletsIds).union(
+                getWalletsUseCase(sdkType = WCM_SDK, chains = chains, excludedIds = WalletConnectModal.excludedWalletsIds, recommendedIds = WalletConnectModal.recommendedWalletsIds).union(
                     getWalletsUseCase(sdkType = WCM_SDK, chains = chains, excludedIds = WalletConnectModal.excludedWalletsIds)
                 ).toList()
             }
-            _modalState.value = WalletConnectModalState(uri, wallets)
+            _modalState.value = WalletConnectModalState(uri, wallets.mapRecentWallet(wcModalStorage.getRecentWalletId()))
         } catch (e: Exception) {
             Timber.e(e)
             _modalState.value = WalletConnectModalState(uri)
         }
     }
+
+    fun updateRecentWalletId(id: String) = with(_modalState) {
+        wcModalStorage.saveRecentWalletId(id)
+        value = value?.copy(wallets = wallets.mapRecentWallet(id))
+    }
 }
+
+private fun List<Wallet>.mapRecentWallet(id: String?) = map {
+    it.apply { it.isRecent = it.id == id }
```

Comment:
You should also be able to do
```suggestion
    it.isRecent = (it.id == id)
```

Reply:
This will return me a list of unit. I would like to have it returned as a list of wallets